src/main/java/net/logstash/logback/composite/CompositeJsonFormatter.java

```diff
@@ -148,23 +133,19 @@ private JsonFactory createJsonFactory() {
                  * when the appender performs the flushes at appropriate times
                  * (such as the end of a batch in the AbstractLogstashTcpSocketAppender).
                  */
-                .disable(JsonGenerator.Feature.FLUSH_PASSED_TO_STREAM);
+                .disable(JsonGenerator.Feature.FLUSH_PASSED_TO_STREAM)
 
+                /*
+                 * Don't let the json generator close the underlying outputStream and let the
+                 * encoder managed it.
+                 */
+                .disable(JsonGenerator.Feature.AUTO_CLOSE_TARGET);
+                
+                //FIXME these settings are mandatory - should we instead apply them *after* jsonGeneratorDecorator is applied (ie. in createGenerator()) ?
```

Comment:
Always mandatory. As you noticed, the two methods the formatter exposes to encode take both a stream or a writer for destination. We cannot let Jackson close them...
This is quite odd tho because the `JsonFactory#createGenerator(stream)` method clearly states that it won't own the stream and therefore won't close it. But since the AUTO_CLOSE_TARGET is enabled by default... the generator will close the stream when itself is closed.

I was thinking that since these two params are "absolutely" required, we'd better to apply them last on the JsonFactory after all other decorators are applied...

Reply:
+1 makes sense to me.